lib/openshift-rest-client.js

```diff
@@ -69,30 +70,67 @@ const spec = JSON.parse(zlib.gunzipSync(fs.readFileSync(path.join(__dirname, 'sp
  * @returns {Promise} Returns the Rest Client
  */
 async function openshiftClient (settings = {}) {
-  let kubeconfig = settings.config;
+  let config = settings.config;
+  const clientConfig = { backend: null, spec, getNames };
 
-  if (kubeconfig) {
+  if (config) {
     // A config is being passed in.  Check if it is an object
-    if (typeof kubeconfig === 'object' && kubeconfig.auth) {
+    if (typeof config === 'object' && config.auth) {
       // Check for the auth username password
-      if ('user' in kubeconfig.auth || 'username' in kubeconfig.auth) {
+      if ('user' in config.auth || 'username' in config.auth) {
         // They are trying the basic auth.
         // Get the access token using the username and password
         // Check to see if we are passing in a username/password
-        const accessToken = await getTokenFromBasicAuth({ insecureSkipTlsVerify: kubeconfig.insecureSkipTlsVerify, url: kubeconfig.url, user: kubeconfig.auth.username || kubeconfig.auth.user, password: kubeconfig.auth.password || kubeconfig.auth.pass, authUrl: kubeconfig.authUrl });
+        const { insecureSkipTlsVerify, url, authUrl } = config;
+        const user = config.auth.username || config.auth.user;
+        const password = config.auth.password || config.auth.pass;
 
-        kubeconfig = {
-          url: kubeconfig.url,
-          auth: {
-            bearer: accessToken
-          },
-          insecureSkipTlsVerify: 'insecureSkipTlsVerify' in kubeconfig ? Boolean(kubeconfig.insecureSkipTlsVerify) : false
+        const accessToken = await getTokenFromBasicAuth({ insecureSkipTlsVerify, url, user, password, authUrl });
+        const clusterUrl = authUrl || url;
+        // Create clusterName from clusterUrl by removing 'https://'
+        const clusterName = clusterUrl.replace(/(^\w+:|^)\/\//, '');
+        config = {
```

Comment:
ummmm,  what is all this stuff?   why are you creating a whole config?  it seems like you are making it unnecessarily more complicated that this actually needs to be

Reply:
Creating a Request without specifying kubeconfig [is deprecated now](https://github.com/godaddy/kubernetes-client/blob/master/backends/request/client.js#L107-L113) and the reason why I added a this kubeconfig template is that it is required by the [converter here](https://github.com/godaddy/kubernetes-client/blob/master/backends/request/config.js#L15)

I didn't find any other way, but will appreciate any suggestion

Reply:
Yeah I didn't find other way either.